### ISSUE TITLE
web-client: Cannot request pre-migration transaction

### ISSUE THREAD

**Issue by sisou** [nimiq/core-rs-albatross#3201]
Requesting a pre-migration transaction [**through the web-client**](https://demo.nimiq.com) for hash `e5ccd2e7b892474b29af0d9fdc5a802989a8e48d77fb6fe85966a3f8fb65049e` which was in block `#3388464`, I get this instead from all peers:

```
await $.client.getTransaction("e5ccd2e7b892474b29af0d9fdc5a802989a8e48d77fb6fe85966a3f8fb65049e")
...
2024-12-17T09:18:25.655000064Z DEBUG consensus_proxy      | Performing txn proof request without block number
2024-12-17T09:18:25.735000064Z DEBUG consensus_proxy      | New txns proof and block from peer peer=... block=#3456000:MA:968f7ad967
2024-12-17T09:18:25.880999936Z DEBUG consensus_proxy      | Error on remote side while requesting block proof error=bad block number 3456000 peer=12D3KooWKTLbkE3MP6EhuXYGbF7i2tVqCwf5ATVc3ffW2gANREbE
...
Error: Transaction not found
```

I can get this transaction via RPC (from a pre-migration-enabled node) just fine.

**Comment by sisou** [comment]
The request handler for the `RequestBlocksProof` message has a [condition that returns the `BadBlockNumber` error for the PoS genesis block](https://github.com/nimiq/core-rs-albatross/blob/15586d0dabe9d9a0b97931662a72fe6030f97cce/consensus/src/messages/handlers.rs#L734).

So the bug seems to be that the web-client requests a proof for that block, when it can't and also shouldn't.

**Comment by sisou** [comment]
Already fixed (but not released) in 8e40a65a10c67edbbc72c7ca6d7ec030f98b87f4 as part of #3187.
